Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -138,38 +170,85 @@ std::map<std::string, std::string> DiscusMultipleScatteringCorrection::validateI
+    1) It's not physical to have a one-sided S(Q,w)
+    2) if S(Q,w) only includes positive w (energy loss) this opens up the possibility of not being able to complete
+       a neutron path through to the detector if the neutron ends up with energy below the min w
+    3) DISCUS took a one sided S(Q,w) and generated values for opposite w using detailed balance so make it clear
+       this algorithm doesn't do that*/
+    if ((axisUnits.find("DeltaE") != axisUnits.end()) && !SQWS->getAxis(1)->isSpectra()) {
+      bool atLeastOnePositive = false;
+      for (size_t iHist = 0; iHist < SQWS->getNumberHistograms(); iHist++) {
+        for (size_t iBin = 0; iBin < SQWS->y(iHist).size(); iBin++) {
+          double wValue;
+          if (SQWS->getAxis(0)->unit()->unitID() == "DeltaE")
+            wValue = SQWS->dataX(0)[iBin];
+          else {
+            wValue = SQWS->getAxis(1)->getValue(iBin);
+          }
+          if ((SQWS->y(iHist)[iBin] > 0.) && (wValue < 0.))
+            atLeastOnePositive = true;
+        }
+      }
+      if (!atLeastOnePositive)
+        issues["StructureFactorWorkspace"] += "S(Q, w) must have some positive values for negative w\n";
+    }
   }
 
-  const std::string emodeStr = getProperty("EMode");
-  auto emode = Kernel::DeltaEMode::fromString(emodeStr);
-  if (emode != Kernel::DeltaEMode::Elastic) {
-    issues["EMode"] = "Only elastic mode is supported at the moment";
+  for (size_t i = 0; i < SQWS->getNumberHistograms(); i++) {
+    auto y = SQWS->y(i);
+    if (std::any_of(y.cbegin(), y.cend(), [](const auto yval) { return yval < 0 || std::isnan(yval); }))
+      issues["StructureFactorWorkspace"] += "S(Q) workspace must have all y >= 0";
   }
 
-  const int nlambda = getProperty("NumberOfWavelengthPoints");
-  if (!isEmpty(nlambda)) {
+  const int nSimulationPoints = getProperty("NumberOfSimulationPoints");
+  if (!isEmpty(nSimulationPoints)) {
     InterpolationOption interpOpt;
     const std::string interpValue = getPropertyValue("Interpolation");
     interpOpt.set(interpValue, false, false);
-    const auto nlambdaIssue = interpOpt.validateInputSize(nlambda);
-    if (!nlambdaIssue.empty()) {
-      issues["NumberOfWavelengthPoints"] = nlambdaIssue;
-    }
+    const auto nSimPointsIssue = interpOpt.validateInputSize(nSimulationPoints);
+    if (!nSimPointsIssue.empty())
+      issues["NumberOfSimulationPoints"] = nSimPointsIssue;
+  }
+
+  const bool simulateEnergiesIndependently = getProperty("SimulateEnergiesIndependently");
+  if (simulateEnergiesIndependently) {
+    if (inputWS->getEMode() == Kernel::DeltaEMode::Elastic)
+      issues["SimulateEnergiesIndependently"] =
+          "SimulateEnergiesIndependently is only applicable to inelastic direct geometry calculations";
+    if (inputWS->getEMode() == Kernel::DeltaEMode::Indirect)
+      issues["SimulateEnergiesIndependently"] =
+          "SimulateEnergiesIndependently is only applicable to inelastic direct geometry calculations. Different "
```

Comment:
I'd separated it out because the message is slightly different for Elastic and Indirect modes. The Indirect message includes an extra bit of explanation that's not relevant to someone running on an Elastic workspace